compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -16843,7 +16843,18 @@ private STNode parseTypedBindingPatternTypeRhs(STNode typeDesc, ParserRuleContex
             case OPEN_BRACKET_TOKEN:
                 // T[..] ..
                 STNode typedBindingPattern = parseTypedBindingPatternOrMemberAccess(typeDesc, true, true, context);
-                assert typedBindingPattern.kind == SyntaxKind.TYPED_BINDING_PATTERN;
+                if (typedBindingPattern.kind != SyntaxKind.TYPED_BINDING_PATTERN) {
```

Comment:
The fix is not correct we need to check why we not getting a binding pattern from `parseTypedBindingPatternOrMemberAccess`. The error handler should insert a variable name after the `,`

Reply:
Yes. The current fix is not appropriate. We need to fix two places. 
1. https://github.com/ballerina-platform/ballerina-lang/blob/6b312cf275684f538aaed692d83c85937dd5b788/compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java#L17442-L17446
Here when the `isTypedBindingPattern = true`, the member cannot be an expression for sure. 
So we need to do following check in `ERROR_CONSTRUCTOR` switch case immediately. 
```java
if (isTypedBindingPattern) {
    return SyntaxKind.LIST_BINDING_PATTERN;
}
```

2. https://github.com/ballerina-platform/ballerina-lang/blob/6b312cf275684f538aaed692d83c85937dd5b788/compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java#L17847-L17848
Here getBindingPattern() validation seem to be missing. It has to be `memberList.add(getBindingPattern(member, true));`

Reply:
Updated